docs_src/design/adr/devops/0010-Release-Artifacts.md

```diff
@@ -0,0 +1,95 @@
+# Release Artifacts
+
+## Status
+
+In Review 06/10/2020
+
+## Context
+
+During the Geneva release of EdgeX Foundry the DevOps WG transformed the CI/CD process with new Jenkins pipeline functionality. After this new functionality was added we also started adding release automation. This new automation is outlined in ADR 0007 Release Automation. However, in ADR 0007 only three release artifact types are outlined. This document is meant to be a living document to try to outline the artifact types for an EdgeX Foundry release and should be modified as that changes.
+
+## Release Artifact Types
+
+### Docker Images
+
+*Tied to Code Release?* Yes
+
+Docker images are released for every named release of EdgeX Foundry. During development the community releases images to the `docker.staging` repository in [Nexus](http://nexus3.edgexfoundry.org). At the time of release we promote the last tested image from `docker.staging` to `docker.release`. In addition to that we will publish the docker image on [DockerHub](https://hub.docker.com/orgs/edgexfoundry).
+
+#### Nexus Retention Policy
+
+##### docker.snapshots
+Retention Policy: 90 days since last download
+
+Contains: Docker images that are not expected to be released. This will contain EdgeX that images are built with each merge and should be available to test things against a specific commit. In addition to this we also build images to optimize the builds in the CI infrastructure.
+
+Docker Tags Used: Git commit, Version-Git commit, Version (ci-build-images only), Latest (ci-build-images-only)
+
+##### docker.staging
+Retention Policy: 180 days since last download
+
+Contains: Docker images built for potential release and testing purposes.
+
+Docker Tags Used: Version (ie: v1.x), Release Branch (master, fuji, etc), Latest
+
+##### docker.release
+Retention Policy: No automatic removal. Requires TSC approval to remove images from this repository.
```

Comment:
So are barcelona images still available? Should we be cleaning these up?

Reply:
TSC can vote to remove any old images. For example you can look at all the versions tagged for `docker-core-config-seed-go` in the release repo here: https://nexus3.edgexfoundry.org/#browse/browse:docker.release:v2%2Fdocker-core-config-seed-go%2Ftags

Reply:
these are good to keep around, useful for historical purposes including testing for regressions.

Reply:
When we wrote the Nexus policy we thought it should be up to the TSC to decide whether or not to clean up old releases.

Reply:
I have the issue of Docker (and Snap) image retention and purging as an Architects' meeting subject.  I think this ADR can proceed while we address this issue.